rslib/src/revlog/mod.rs

```diff
@@ -59,6 +59,16 @@ pub struct RevlogEntry {
     pub review_kind: RevlogReviewKind,
 }
 
+pub trait RemoveBefore {
+    fn remove_before(self, ms: i64) -> Self;
+}
+
+impl RemoveBefore for Vec<RevlogEntry> {
+    fn remove_before(self, ms: i64) -> Self {
+        self.into_iter().filter(|revlog| revlog.id.0 > ms).collect()
+    }
+}
+
```

Comment:
I'd prefer this were a standalone function instead of a method in a trait, as we only use it twice. It would also be nice if it avoided doing anything when ms is 0, as otherwise the we wastefully rebuild the vec without doing any filtering.